File: LGU_Solano/HRMIS-Solano/HRMIS-Solano/Connection.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using MySql.Data.MySqlClient;
using System.IO;
using Microsoft.VisualBasic.Devices;
using System.Security.Cryptography;

namespace HRMIS_Solano
{
    class Connection
    {
        public static MySqlConnection Conn = new MySqlConnection();
        // This constant string is used as a "salt" value for the PasswordDeriveBytes function calls.
        // This size of the IV (in bytes) must = (keysize / 8).  Default keysize is 256, so the IV must be
        // 32 bytes long.  Using a 16 character string here gives us 32 bytes when converted to a byte array.
        private const string initVector = "tu89geji340t89u2";

        // This constant is used to determine the keysize of the encryption algorithm.
        private const int keysize = 256;
        public void SetConstr()
        {
            Computer c = new Computer();
            string path = c.FileSystem.SpecialDirectories.ProgramFiles + @"\HRMIS Config\ConfigurationHRMIS.exe";
            StreamReader sr = new StreamReader(path);
            string constr = sr.ReadLine();
            sr.Close();
            constr = Decrypt(constr, "H@CK3R");
            Conn = new MySqlConnection(constr);
        }
        public static string Decrypt(string cipherText, string passPhrase)
        {
            byte[] initVectorBytes = Encoding.ASCII.GetBytes(initVector);
            byte[] cipherTextBytes = Convert.FromBase64String(cipherText);
            PasswordDeriveBytes password = new PasswordDeriveBytes(passPhrase, null);
            byte[] keyBytes = password.GetBytes(keysize / 8);
            RijndaelManaged symmetricKey = new RijndaelManaged();
            symmetricKey.Mode = CipherMode.CBC;
            ICryptoTransform decryptor = symmetricKey.CreateDecryptor(keyBytes, initVectorBytes);
            MemoryStream memoryStream = new MemoryStream(cipherTextBytes);
            CryptoStream cryptoStream = new CryptoStream(memoryStream, decryptor, CryptoStreamMode.Read);
            byte[] plainTextBytes = new byte[cipherTextBytes.Length];
            int decryptedByteCount = cryptoStream.Read(plainTextBytes, 0, plainTextBytes.Length);
            memoryStream.Close();
            cryptoStream.Close();
            return Encoding.UTF8.GetString(plainTextBytes, 0, decryptedByteCount);
        }
    }
}

File: LGU_Solano/MDF/MDFIS/Connection.vb
'Import Libraries
Imports System.IO
Imports Microsoft.VisualBasic.Devices
Imports System.Security.Cryptography
Imports System.Text
Imports MySql.Data.MySqlClient
Public Class Connection

    'Declare Connection
    Private DES As New TripleDESCryptoServiceProvider
    Private MD5 As New MD5CryptoServiceProvider
    Public cnstr As MySqlConnection

    Public Sub SetConstr()
        Dim c As Computer = New Computer()
        Dim path As String = c.FileSystem.SpecialDirectories.ProgramFiles + "\MDFIS Config\ConfigurationMDFIS.exe"
        Dim sr As StreamReader = New StreamReader(path)
        Dim constr As String = sr.ReadLine()
        sr.Close()
        constr = Decrypt(constr, "H@CK1Ng")
        cnstr = New MySqlConnection(constr)
    End Sub
    Public Function MD5Hash(ByVal value As String) As Byte()
        Return MD5.ComputeHash(ASCIIEncoding.ASCII.GetBytes(value))
    End Function

    Public Function Decrypt(ByVal encryptedString As String, ByVal key As String) As String
        Dim value As String = ""
        Try

            DES.Key = MD5Hash(key)
            DES.Mode = CipherMode.ECB
            Dim Buffer As Byte() = Convert.FromBase64String(encryptedString)
            value = ASCIIEncoding.ASCII.GetString(DES.CreateDecryptor().TransformFinalBlock(Buffer, 0, Buffer.Length))
        Catch ex As Exception
            MessageBox.Show("Invalid Key", "Decryption Failed", MessageBoxButtons.OK, MessageBoxIcon.Exclamation)
        End Try
        Return value
    End Function

End Class
